### PR TITLE
Mouse Interactions

### PR DESCRIPTION
**Pull request by ghost**
# New feature description
- Upgraded Babylon.js to 5.2.0 in order to use the new `zoomToMouseLocation` feature on the ArcRotateCamera
- Updated the camera to orbit around the model even after zooming or panning

# Checklist

-   [x] All acceptance criteria are met.
-   [x] Relevant documentation, if any, has been written/updated.
-   [x] The changelog has been updated, if applicable.
-   [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

https://www.loom.com/share/06f28618724541119513b9119ba09e39

### CODE REVIEW

**Comment by braustin20** [comment]
I'm having trouble compiling this locally. Getting an eslint error. Have you seen this? Perhaps try an `npm run clean` and then `npm run setup` to test.

If you aren't seeing this perhaps it's my local environment.

```
Html Webpack Plugin:
  Error: Child compilation failed:
  Must use import to load ES Module: /Users/braustin/Dev/product-viewer/packages  /viewer-demos/node_modules/react-scripts/node_modules/@eslint/eslintrc/univers  al.js
  require() of ES modules is not supported.
  require() of /Users/braustin/Dev/product-viewer/packages/viewer-demos/node_mod  ules/react-scripts/node_modules/@eslint/eslintrc/universal.js from /Users/brau  stin/Dev/product-viewer/packages/viewer-demos/node_modules/react-scripts/node_  modules/eslint/lib/linter/linter.js is an ES module file as it is a .js file w  hose nearest parent package.json contains "type": "module" which defines all .  js files in that package scope as ES modules.
  Instead rename universal.js to end in .cjs, change the requiring code to use i  mport(), or remove "type": "module" from /Users/braustin/Dev/product-viewer/pa  ckages/viewer-demos/node_modules/react-scripts/node_modules/@eslint/eslintrc/p  ackage.json.
  
  - child-compiler.js:169 
    [viewer-demos]/[html-webpack-plugin]/lib/child-compiler.js:169:18
  
  - Compiler.js:559 
    [viewer-demos]/[webpack]/lib/Compiler.js:559:11
  
  - Compiler.js:1129 
    [viewer-demos]/[webpack]/lib/Compiler.js:1129:17
  
  
  - task_queues.js:97 processTicksAndRejections
    internal/process/task_queues.js:97:5

File: packages/product-viewer/src/types/polyfills.d.ts [hunk 1]
```diff
@@ -24,5 +24,6 @@ export {};
 declare global {
 	interface Window {
 		ResizeObserver: any;
+		MSStream: any;
```
Comment (braustin20) [inline packages/product-viewer/src/types/polyfills.d.ts:27]: What is this used for?

Comment (ghost) [reply packages/product-viewer/src/types/polyfills.d.ts:27]: It looks like that type was removed from a recent version of typescript.. without it I get 
```
src/tools/Utils.ts:50:58 - error TS2339: Property 'MSStream' does not exist on type 'Window & typeof globalThis'.

50   (/iPad|iPhone|iPod/.test(navigator.userAgent) && !self.MSStream) ||
                                                            ~~~~~~~~
```
Also just it listed as removed here: https://github.com/microsoft/TypeScript-DOM-lib-generator/issues/1029

**Comment by ghost** [comment]
> I'm having trouble compiling this locally. Getting an eslint error. Have you seen this? Perhaps try an `npm run clean` and then `npm run setup` to test.
> 
> If you aren't seeing this perhaps it's my local environment.
> 
> ```
> Html Webpack Plugin:
>   Error: Child compilation failed:
>   Must use import to load ES Module: /Users/braustin/Dev/product-viewer/packages  /viewer-demos/node_modules/react-scripts/node_modules/@eslint/eslintrc/univers  al.js
>   require() of ES modules is not supported.
>   require() of /Users/braustin/Dev/product-viewer/packages/viewer-demos/node_mod  ules/react-scripts/node_modules/@eslint/eslintrc/universal.js from /Users/brau  stin/Dev/product-viewer/packages/viewer-demos/node_modules/react-scripts/node_  modules/eslint/lib/linter/linter.js is an ES module file as it is a .js file w  hose nearest parent package.json contains "type": "module" which defines all .  js files in that package scope as ES modules.
>   Instead rename universal.js to end in .cjs, change the requiring code to use i  mport(), or remove "type": "module" from /Users/braustin/Dev/product-viewer/pa  ckages/viewer-demos/node_modules/react-scripts/node_modules/@eslint/eslintrc/p  ackage.json.
>   
>   - child-compiler.js:169 
>     [viewer-demos]/[html-webpack-plugin]/lib/child-compiler.js:169:18
>   
>   - Compiler.js:559 
>     [viewer-demos]/[webpack]/lib/Compiler.js:559:11
>   
>   - Compiler.js:1129 
>     [viewer-demos]/[webpack]/lib/Compiler.js:1129:17
>   
>   
>   - task_queues.js:97 processTicksAndRejections
>     internal/process/task_queues.js:97:5
> ```

Odd.. are you running Node 12? I was able to `clean`, `setup` and `build` just fine..
```
> node -v       
v12.22.7
```

**Comment by braustin20** [comment]
> > I'm having trouble compiling this locally. Getting an eslint error. Have you seen this? Perhaps try an `npm run clean` and then `npm run setup` to test.
> > If you aren't seeing this perhaps it's my local environment.
> > ```
> > Html Webpack Plugin:
> >   Error: Child compilation failed:
> >   Must use import to load ES Module: /Users/braustin/Dev/product-viewer/packages  /viewer-demos/node_modules/react-scripts/node_modules/@eslint/eslintrc/univers  al.js
> >   require() of ES modules is not supported.
> >   require() of /Users/braustin/Dev/product-viewer/packages/viewer-demos/node_mod  ules/react-scripts/node_modules/@eslint/eslintrc/universal.js from /Users/brau  stin/Dev/product-viewer/packages/viewer-demos/node_modules/react-scripts/node_  modules/eslint/lib/linter/linter.js is an ES module file as it is a .js file w  hose nearest parent package.json contains "type": "module" which defines all .  js files in that package scope as ES modules.
> >   Instead rename universal.js to end in .cjs, change the requiring code to use i  mport(), or remove "type": "module" from /Users/braustin/Dev/product-viewer/pa  ckages/viewer-demos/node_modules/react-scripts/node_modules/@eslint/eslintrc/p  ackage.json.
> >   
> >   - child-compiler.js:169 
> >     [viewer-demos]/[html-webpack-plugin]/lib/child-compiler.js:169:18
> >   
> >   - Compiler.js:559 
> >     [viewer-demos]/[webpack]/lib/Compiler.js:559:11
> >   
> >   - Compiler.js:1129 
> >     [viewer-demos]/[webpack]/lib/Compiler.js:1129:17
> >   
> >   
> >   - task_queues.js:97 processTicksAndRejections
> >     internal/process/task_queues.js:97:5
> > ```
> 
> Odd.. are you running Node 12? I was able to `clean`, `setup` and `build` just fine..
> 
> ```
> > node -v       
> v12.22.7
> ```

Ah yep it seems to happen on v12.16.2. I installed v12.22.7 using nvm and it worked. Went back to 12.16.2 and got the same error. Must be some updated dependency causing this.

**Review by braustin20 — APPROVED** [review]

**Pull request merged**